File: LeetCode/Java Solutions/Sort Integers by The Number of 1 Bits/sort.java
import java.util.Arrays;

class Solution {
    public int[] sortByBits(int[] arr) {
        // Sorting using custom comparator
        Integer[] boxedArr = Arrays.stream(arr).boxed().toArray(Integer[]::new);
        Arrays.sort(boxedArr, (a, b) -> {
            int countA = Integer.bitCount(a);
            int countB = Integer.bitCount(b);
            if (countA == countB) return a - b;
            return countA - countB;
        });
        return Arrays.stream(boxedArr).mapToInt(Integer::intValue).toArray();
    }
}

File: LeetCode/Python Solutions/Sort Integers by The Number of 1 Bits/sort.py
class Solution:
    def sortByBits(self, arr: 'List[int]') -> 'List[int]':
        # Helper function to count 1's in binary representation
        def countOnes(num):
            return bin(num).count('1')

        # Sorting using lambda function as custom comparator
        return sorted(arr, key=lambda x: (countOnes(x), x))


